src/upchunk.ts

```diff
@@ -483,26 +483,35 @@ export class UpChunk {
    * Send chunk of the file with appropriate headers
    */
   protected async sendChunk(chunk: Blob) {
-    const rangeStart = this.nextChunkRangeStart;
-    const rangeEnd = rangeStart + chunk.size - 1;
-    const headers = {
-      ...this.headers,
-      'Content-Type': this.file.type,
-      'Content-Range': `bytes ${rangeStart}-${rangeEnd}/${this.file.size}`,
+    const getHeaders = async (): Promise<XhrHeaders> => {
```

Comment:
non-blocking: this could potentially be refactored to use await on the header method.

Something like:
```js
const extraHeaders = await typeof this.headers === 'function' ? this.headers() : this.headers;
```
while keeping all the other code the same in the same location. (And spreading `...extraHeaders` instead of `...this.headers`)